.github/workflows/get-latest-posts.yml

```diff
@@ -21,16 +21,12 @@ jobs:
           npm install
           node --experimental-fetch bin/generateDevToPosts.js
           node bin/generateHashnodeUrlMapping.js
-      - name: Commit changes
-        id: commit
+      - name: Setup git config
+        run: |
+          git config user.name 'token-generator-app[bot]'
+          git config user.email '82042599+token-generator-app[bot]@users.noreply.github.com'
+      - name: PR for Blog Post Updates
         env:
           GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
         run: |
-          git config user.name "GitHub Actions Bot"
-          git config user.email "<>"
-          git pull origin main
-          git add .
-          if [[ -n "$(git status --porcelain)" ]]; then
-            git commit -m "chore (automated): update blog posts"
-            git push origin main
-          fi
+          ./bin/pr-blog-posts.sh
```

Comment:
It was easier to put this all in a script instead. Same for videos below. It could be one script and I pass in something to say video or blog post, but I'm keeping it simple for now and a little duplication isn't a biggie.